### Data Analysis with Panda
- Bike Sales
- Michael Hendricks
- July 20

In [135]:
import pandas as pd
import numpy as np

In [136]:
station_df = pd.read_csv('data/station.csv')
trip_df = pd.read_csv('data/trip.csv', error_bad_lines=False)
weather_df = pd.read_csv('data/weather.csv')

b'Skipping line 50794: expected 12 fields, saw 20\n'


In [156]:
trip_df[['tripduration']].mean()

tripduration    1178.295675
dtype: float64

### 1. The average trip duration for a borrowed bicycle is 1178.295675

In [157]:
trip_df['birthyear'].mode()

0    1987.0
dtype: float64

### 2. The most common birth year of a bicycle sharer is 1987, and since that's the year I was born, that makes them 33

In [158]:
#weather_df['Date'] = pd.to_datetime(weather_df['Date'], format='%m/%d/%Y')
#weather_df.loc[weather_df['Date'].dt.month == 12][['Date', 'Precipitation_In']].mean()

In [159]:
prec_per_day = weather_df[['Date','Precipitation_In']].mean()

In [160]:
prec_per_month = prec_per_day * 30

In [161]:
prec_per_month

Precipitation_In    3.151959
dtype: float64

### 3. The average precipitation per month is 3.15

In [165]:
#station_df[['name','current_dockcount']]
station_df[['current_dockcount']].mean()

current_dockcount    16.517241
dtype: float64

### 4. The average number of bikes at a given station is 16.517241

In [166]:
# notnull() filters out NaN returns
modified_stations = station_df[station_df['modification_date'].notnull()][['name','install_dockcount', 'modification_date', 'current_dockcount']].set_index('name')

In [168]:
modified_stations[['install_dockcount']].mean()

install_dockcount    17.529412
dtype: float64

In [169]:
modified_stations[['current_dockcount']].mean()

current_dockcount    13.764706
dtype: float64

### 5. When a bike station is modified it is more likely that it's lose bikes because the average dock count of stations before modifications was higher than the average dock count of those same stations after modifications

In [182]:
weather_df.head()

Date  Max_Temperature_F  Mean_Temperature_F  Min_TemperatureF  \
0  10/13/2014                 71                62.0                54   
1  10/14/2014                 63                59.0                55   
2  10/15/2014                 62                58.0                54   
3  10/16/2014                 71                61.0                52   
4  10/17/2014                 64                60.0                57   

   Max_Dew_Point_F  MeanDew_Point_F  Min_Dewpoint_F  Max_Humidity  \
0               55               51              46            87   
1               52               51              50            88   
2               53               50              46            87   
3               49               46              42            83   
4               55               51              41            87   

   Mean_Humidity  Min_Humidity  ...  Mean_Sea_Level_Pressure_In  \
0             68            46  ...                       29.79   
1             78            63  ...                       29.75   
2             77            67  ...                       29.71   
3             61            36  ...                       29.95   
4             72            46  ...                       29.78   

   Min_Sea_Level_Pressure_In  Max_Visibility_Miles  Mean_Visibility_Miles  \
0                      29.65                    10                     10   
1                      29.54                    10                      9   
2                      29.51                    10                      9   
3                      29.81                    10                     10   
4                      29.73                    10                     10   

   Min_Visibility_Miles  Max_Wind_Speed_MPH  Mean_Wind_Speed_MPH  \
0                     4                  13                    4   
1                     3                  10                    5   
2                     3                  18                    7   
3                    10                   9                    4   
4                     6                   8                    3   

   Max_Gust_Speed_MPH Precipitation_In  Events  
0                  21             0.00    Rain  
1                  17             0.11    Rain  
2                  25             0.45    Rain  
3                   -             0.00    Rain  
4                   -             0.14    Rain  

[5 rows x 21 columns]

In [184]:
weather_df[['Max_Temperature_F']].max()

Max_Temperature_F    98
dtype: int64

### 6. The hottest day on record hit a max of 98 degrees F

In [186]:
weather_df[['Min_TemperatureF']].min()

Min_TemperatureF    23
dtype: int64

### 7. The coldest day on record got as cold as 23 defrees F

In [187]:
weather_df[['Mean_Visibility_Miles']].mean()

Mean_Visibility_Miles    9.429608
dtype: float64

### 8. The overall avg visibility is 9.42 miles